proxy/src/connection.rs

```diff
@@ -111,7 +111,7 @@ impl BoundPort {
     // TLS when needed.
     pub fn listen_and_fold<T, F, Fut>(
         self,
-        tls: Option<(tls::Identity, tls::ServerConfigWatch)>,
+        tls: tls::ConditionalConnectionConfig<tls::ServerConfigWatch>,
```

Comment:
NIT/TIOLI: this type name is pretty long; maybe the `tls` module should expose aliases like
```rust
pub type ConditionalServerConfig = ConditionalConnectionConfig<ServerConfigWatch>;
pub type ConditionalClientConfig = ConditionalConnectionConfig<ClientConfigWatch>;
```